lib/ASRouterTriggerListeners.jsm

```diff
@@ -33,6 +33,29 @@ function isPrivateWindow(win) {
   return !(win instanceof Ci.nsIDOMWindow) || win.closed || PrivateBrowsingUtils.isWindowPrivate(win);
 }
 
+/**
+ * Check current location against the list of whitelisted hosts
+ * Additionally verify for redirects and check original request URL against
+ * the whitelist.
+ *
+ * @returns {string} - the host that matched the whitelist
```

Comment:
I went back on this and now returning the `host` that matched the whitelist because we have an additional check 
https://github.com/mozilla/activity-stream/blob/b89bae21fd28625f3d70cadb4f1a404340aeb129/lib/ASRouterTargeting.jsm#L341
Where ensure that the selected message matches the trigger. If we would have passed down `aLocationURI.host` that could be `redirect.foo.com` and not match `foo.com` even if `checkHost` initially returned `true`.